build-tools/packages/build-cli/src/lib/package.ts

```diff
@@ -0,0 +1,453 @@
+ *
+ * This function exclusively uses the tags in the repo to determine whether a release has bee done or not.
+ *
+ * @internal
+ */
+export async function isReleased(
+    context: Context,
+    releaseGroupOrPackage: MonoRepo | Package | string,
+    version: string,
+    log?: Logger,
+): Promise<boolean> {
+    await context.gitRepo.fetchTags();
+
+    const tagName = generateReleaseTagName(releaseGroupOrPackage, version);
+    if (typeof releaseGroupOrPackage === "string" && isReleaseGroup(releaseGroupOrPackage)) {
+        // eslint-disable-next-line no-param-reassign
+        releaseGroupOrPackage = context.repo.releaseGroups.get(releaseGroupOrPackage)!;
+    }
+
+    log?.verbose(`Checking for tag '${tagName}'`);
+    const rawTag = await context.gitRepo.getTags(tagName);
+    return rawTag.trim() === tagName;
+}
+
+/**
+ * Generates the correct tag name for the release of a given release group and version.
+ *
+ * @param releaseGroupOrPackage - The release group or independent package to generate a tag name for.
+ * @param version - The version to use for the generated tag.
+ * @returns The generated tag name.
+ *
+ * @internal
+ */
+export function generateReleaseTagName(
+    releaseGroupOrPackage: MonoRepo | Package | string,
+    version?: string,
+): string {
+    let tagName = "";
+
+    if (releaseGroupOrPackage instanceof MonoRepo) {
+        const kindLowerCase = releaseGroupOrPackage.kind.toLowerCase();
+        tagName = `${kindLowerCase}_v${version ?? releaseGroupOrPackage.version}`;
+    } else if (releaseGroupOrPackage instanceof Package) {
+        tagName = `${getPackageShortName(releaseGroupOrPackage.name)}_v${
+            version ?? releaseGroupOrPackage.version
+        }`;
+    } else {
+        tagName = `${getPackageShortName(releaseGroupOrPackage)}_v${version}`;
+    }
+
+    return tagName;
+}
+
+/**
+ *  Returns an array of all the git tags associated with a release group.
+ *
+ * @param context - The {@link Context}.
+ * @param releaseGroupOrPackage - The release group or independent package to get tags for.
+ * @returns An array of all all the tags for the release group or package.
+ *
+ * @internal
+ */
+export async function getTagsForReleaseGroup(
+    context: Context,
+    releaseGroupOrPackage: ReleaseGroup | ReleasePackage,
+): Promise<string[]> {
+    let prefix = "";
+    try {
+        prefix = isReleaseGroup(releaseGroupOrPackage)
+            ? releaseGroupOrPackage.toLowerCase()
+            : getPackageShortName(releaseGroupOrPackage);
+    } catch {
+        console.log(releaseGroupOrPackage);
+    }
+
+    const allTags = await context.gitRepo.getAllTags(`${prefix}*`);
+    return allTags;
+}
+
+/**
+ * Parses the version from a git tag.
+ *
+ * @param tag - The tag.
+ * @returns - The version string, or undefined if one could not be found.
+ *
+ * @internal
+ */
+export function getVersionFromTag(tag: string): string | undefined {
+    // TODO: Consider a stronger check here using a regex.
```

Comment:
Did you intend to do this before merging, or later?

Reply:
Not in this PR, because I think it's more likely that I'll be too conservative with the regex, and with our tags right now the risk of the current code failing are low.